### ISSUE TITLE
os: missing conversion of internal/poll.ErrFileClosing to fs.ErrClosed

### ISSUE THREAD

**Issue by nolta** [golang/go#58622]
TL;DR: Posted to go-nuts, and Ian Lance Taylor [responded](https://groups.google.com/g/golang-nuts/c/NIbQu4gg4CQ/m/_A2X4zTuBgAJ):

> I think this is a bug. (*File).readFrom, or perhaps NewSyscallError, 
should convert poll.ErrFileClosing to os.ErrClosed, just as 
(*File).wrapErr does.

I believe they're referring to [this line in `os/readfrom_linux.go`](https://cs.opensource.google/go/go/+/master:src/os/readfrom_linux.go;l=45;drc=0844ff8eef81e124c1fecba82dd5843745427fa4).

### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.18.10 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Don't know.

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/runner/.cache/go-build"
GOENV="/home/runner/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/runner/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/runner/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/opt/hostedtoolcache/go/1.18.10/x64"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/opt/hostedtoolcache/go/1.18.10/x64/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18.10"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/runner/work/gosh/gosh/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build918714692=/tmp/go-build -gno-record-gcc-switches"</pre></details>

### What did you do?

My code is essentially running shell pipelines, and i want to catch harmless errors where a process on the left side of the pipeline dies because the right side has finished. Think `{ echo 1; echo 2; echo 3; } | head -n1`.

So in addition to catching EPIPEs and SIGPIPEs, i catch file closed errors:

```go
// ignore writes to closed pipe
switch err := err.(type) {
case *fs.PathError:
    if err.Op == "write" && err.Path == "|1" && errors.Is(err.Err, fs.ErrClosed) {
        return nil
    }
}
return err
```

### What did you expect to see?

Nothing.

### What did you see instead?

Code dies with the error message `write |1: copy_file_range: use of closed file`. Since this error appears to be [internal/poll.ErrFileClosing](https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/internal/poll/fd.go;l=35), i can't catch it since i can't import the package.
